packages/plugin/src/VisualizationPlugin.js

```diff
@@ -54,8 +78,26 @@ export const VisualizationPlugin = ({
         [engine]
     )
 
+    const doFetchOuLevelsData = useCallback(async () => {
+        const ouLevelsData = await apiFetchOrganisationUnitLevels(engine)
+
+        return ouLevelsData.orgUnitsLevels.organisationUnitLevels
+    }, [engine])
+
+    useEffect(() => {
+        const doFetch = async () => {
+            const orgUnitLevels = await doFetchOuLevelsData()
+
+            setOuLevels(orgUnitLevels)
+        }
+
+        doFetch().catch(error => onError(error))
+        /* eslint-disable-next-line react-hooks/exhaustive-deps */
+    }, [])
```

Comment:
This should be `useDataQuery` instead of `dataEngine`.  The line `const [ouLevels, setOuLevels] = useState(null)` and all of this block should be replaced with:

```js
const { data: ouLevelsResponse } = useDataQuery(orgUnitLevelsQuery, { onError })
const ouLevels = ouLevelsResponse?.orgUnitsLevels.organisationUnitLevels
```

There is a slight functional difference here - if the component unmounts before the request completes with the code as written it will try to update the state of an unmounted component, whereas `useDataQuery` will abort the request.

Reply:
Fixed.